src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/DateHeaderValueManager.cs

```diff
@@ -107,62 +89,32 @@ public void Dispose()
         /// </summary>
         private void StartTimer()
         {
-            var now = _systemClock.UtcNow;
-            SetDateValues(now);
-
-            if (!_isDisposed)
-            {
-                lock (_timerLocker)
-                {
-                    if (!_timerIsRunning && _dateValueTimer != null)
-                    {
-                        _timerIsRunning = true;
-                        _dateValueTimer.Change(_timerInterval, _timerInterval);
-                    }
-                }
-            }
+            SetDateValues(_systemClock.UtcNow);
+            _timerIsRunning = !_isDisposed;
         }
 
-        /// <summary>
-        /// Stops the timer
-        /// </summary>
-        private void StopTimer()
+        // Called by the Timer (background) thread
+        public void Tick(DateTimeOffset now)
         {
-            if (!_isDisposed)
+            if (_timerIsRunning)
             {
-                lock (_timerLocker)
+                SetDateValues(now);
+
+                if (_hadRequestsSinceLastTimerTick)
```

Comment:
Does `_hadRequestsSinceLastTimerTick` make sense here? We are only preventing header value from encoding when no requests are hitting server, hardly worth additional logic.